crates/sui-types/src/transaction.rs

```diff
@@ -96,6 +96,8 @@ pub enum ObjectArg {
         initial_shared_version: SequenceNumber,
         mutable: bool,
     },
+    // A Move object that can be received in this transaction.
+    Receiving(ObjectRef),
```

Comment:
this will require a protocol config feature flag

to enforce the flag, you'll have to add verification methods to ProgrammableTransaction to check whether this arg is present. You can start from here https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/transaction.rs#L216